syntax = "proto3";

package TSCached;


//一行数据
message Metrics{
  int64 timeStamp = 1;
  map<string,double> fields=2;
}
message Point{
  string dataBase = 1;
  string tableName = 2;
  map<string,string> tags = 3;
  repeated Metrics metrics = 4;
}


message QueryRequest {
  string dataBase = 1;
  string tableName = 2;
  map<string,string> tags = 3;
  int64 startTime = 4;   //开始时间
  int64 endTime = 5;     //结束时间
}

message QueryResponse {
  string dataBase = 1;
  string tableName = 2;
  map<string,string> tags = 3;
  repeated Metrics metrics = 4;
}


message WriteRequest{
  repeated Point points = 1;
}

enum StatusCode {
  OK = 0;
  ERROR = 1;
}

message WriteResponse {
  string message = 1;
  StatusCode statusCode = 2;
}

service TSCachedService{
  rpc QueryPoints(QueryRequest) returns (QueryResponse) {};
  rpc WritePoints(WriteRequest) returns (WriteResponse) {};
}

